File: openapi-test-java-server-spring-client-jdk/testapi/model/PetId.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonValue;

/** Unique pet identifier */
public record PetId(@JsonValue String value) {
  public PetId withValue(String value) {
    return new PetId(value);
  };

  @Override
  public String toString() {
    return value;
  };

  /** Converts a String to this type for Spring @PathVariable binding */
  static public PetId valueOf(String s) {
    return new PetId(s);
  };
}

File: openapi-test-kotlin-server-spring-client-jdk/testapi/model/PetId.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

/** Unique pet identifier */
data class PetId @JsonCreator constructor(@get:JsonValue val value: String) {
  override fun toString(): String {
    return value
  }

  companion object {
    /** Converts a String to this type for Spring @PathVariable binding */
    fun valueOf(s: String): PetId = PetId(s)
  }
}